## 1
#### Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

In [18]:
from math import factorial

In [19]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n-k)))

In [20]:
combinations(52, 4)

270725

Число сочетаний из 52 по 4 равно:

$C_{52}^4 = 270725$

Это общее число вариантов выбрать 4 карты из 52

In [21]:
combinations(13,4)

715

In [22]:
715/270725

0.0026410564225690276

Всего вариантов выбрать из 13 крестей 4 любые карты равно:

$C_{13}^4 = 715$

Соответственно, это исходы, удовлетворяющие условию, значит ответ на первый вопрос:

$\frac{715}{270725} = 0.0026 \approx 0.3\%$

In [23]:
combinations(4,0)

1

In [24]:
combinations(48, 4)

194580

In [25]:
(1*194580)/270725

0.7187367254594146

Вероятность, что ни одна из карты не окажется тузом равна:

$\frac{C_{4}^0 \times C_{48}^4}{C_{13}^4} \approx 0.72 $

Соответственно, вероятность, что хотя бы одна из карт будет туз, равна:

$1 - 0.72 = 0.28$

## 2
#### На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

In [26]:
combinations(10, 3)

120

In [27]:
1/120

0.008333333333333333

Всего вариантов нажать одновременно 3 кнопки:

$C_{10}^3$ = 120

Из них только одна комбинация верная, вероятность угадать ее с первого раза равна:

$\frac{1}{120} \approx 0.008 = 0.8\%$

### 3
#### В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

In [28]:
combinations(9,3) / combinations(15,3)

0.18461538461538463

$\frac{C_{9}^3}{C_{15}^3} \approx 0.18 = 18\%$

### 4
#### В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [29]:
combinations(100, 2)

4950

Всего вариантов купить 2 билета из 100:

$C_{100}^2 = 4950$

Всего существует одно сочетание выигрышных билетов, поэтому вероятность приобрести оба выигрышных билета равна:

$\frac{1}{C_{100}^2} \approx 0.0002 = 0.02\%$